test/test_testing.py

```diff
@@ -683,5 +685,185 @@ def test_calculate_2_shards_against_optimal_shards(self):
                 self.assertEqual(sorted_tests, sorted_shard_tests)
 
 
+class TestAsserts(TestCase):
+    @staticmethod
+    def _assert_fns(device=None):
+        return [torch.testing.assert_tensors_equal, torch.testing.assert_tensors_close]
```

Comment:
Not sure why the device parameter is needed here. Is there a better way to add helper functions to a test case that will be used in `instantiate_device_type_tests()`?

Reply:
You shouldn't need the device parameter here. Here's an example helper method in another device generic test suite:

https://github.com/pytorch/pytorch/blob/f5d6b90c3573a0e5507e4e100569924f6b2ba7f4/test/test_unary_ufuncs.py#L248

Reply:
This helper probably has the exact same problem as I have. It has the `dtype` parameter without actually using it. If I remove it and try to run the test case, I'm hit with

```
Traceback (most recent call last):
  File "/home/philip/git/pytorch/torch/test/test_unary_ufuncs.py", line 353, in test_reference_numerics_extremal
    self._test_reference_numerics(dtype, op, tensors, equal_nan)
  File "/home/philip/git/pytorch/torch/test/test_unary_ufuncs.py", line 309, in _test_reference_numerics
    _helper_reference_numerics(expected, actual, msg, exact_dtype, equal_nan)
  File "/home/philip/git/pytorch/torch/test/test_unary_ufuncs.py", line 287, in _helper_reference_numerics
    self.assertEqualHelper(actual, expected, msg, dtype=dtype, equal_nan=equal_nan, exact_dtype=exact_dtype)
  File "/home/philip/git/pytorch/torch/test/test_unary_ufuncs.py", line 270, in assertEqualHelper
    **kwargs)
TypeError: assertEqual() got an unexpected keyword argument 'dtype'
```

I suspect the test creation simply adds the `dtype` (or `device`) parameter to every method instead of just the ones prefixed by `_test`. I'll investigate.

Reply:
Scratch the above. The problem is that the method is defined as `@staticmethod` but the instantiation adds it back as normal.